src/Trakx.Copper.ApiClient.Tests/Unit/ApiKeyCredentialsProviderTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.Options;
+using NSubstitute;
+using Trakx.Copper.ApiClient.Utils;
+using Trakx.Utils.DateTimeHelpers;
+using Trakx.Utils.Extensions;
+using Xunit;
+
+namespace Trakx.Copper.ApiClient.Tests.Unit
+{
+    public class ApiKeyCredentialsProviderTests
+    {
+        private readonly ApiKeyCredentialsProvider _apiKeyCredentials;
+        private readonly CopperApiConfiguration _configuration;
+        private readonly DateTimeOffset _timestamp;
+        public ApiKeyCredentialsProviderTests()
+        {
+            _configuration = new CopperApiConfiguration { ApiKey = "apikeyPublic", ApiSecret = "c2FsdXRhdG91cw==", BaseUrl = "http://baseUrl.com" };
+            var dateTimeProvider = Substitute.For<IDateTimeProvider>();
+            _timestamp = DateTimeOffset.FromUnixTimeMilliseconds(123456789);
+            dateTimeProvider.UtcNowAsOffset.ReturnsForAnyArgs(_timestamp);
+            var options = Substitute.For<IOptions<CopperApiConfiguration>>();
+            options.Value.ReturnsForAnyArgs(_configuration);
+            _apiKeyCredentials = new ApiKeyCredentialsProvider(options, dateTimeProvider);
+        }
+
+        [Fact]
+        public void Header_should_be_as_expected()
+        {
+            var message = new HttpRequestMessage(HttpMethod.Get, new Uri(_configuration.BaseUrl + "/path1/path2"));
+            _apiKeyCredentials.AddCredentials(message);
+
+            message.Headers.Authorization!.Scheme.Should().Be(ApiKeyCredentialsProvider.ApiKeyHeader);
+            message.Headers.Authorization!.Parameter.Should().Be(_configuration.ApiKey);
+            message.Headers.GetValues(ApiKeyCredentialsProvider.ApiTimestampHeader).Single()
+                .Should().Be(_timestamp.ToUnixTimeMilliseconds().ToString());
+            message.Headers.GetValues(ApiKeyCredentialsProvider.ApiSignatureHeader).Single().Should()
+                .NotBeNullOrEmpty();
+        }
+
+        [Fact]
+        public async Task Signature_should_be_as_expected()
+        {
+            var message = new HttpRequestMessage(HttpMethod.Get, new Uri(_configuration.BaseUrl + "/path1/path2")) { Content = new StringContent("Hello, I'm the body of the request") };
+            _apiKeyCredentials.AddCredentials(message);
+
+            var retrievedSignature = message.Headers.GetValues(ApiKeyCredentialsProvider.ApiSignatureHeader).Single();
+            await VerifySignature(message, retrievedSignature).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public void Signature_should_depends_of_message_content()
+        {
```

Comment:
that check is actually encoded in the `Signature_should_be_as_expected` right?

Reply:
I've made this test to prove that the signature depends on the content, but yes the `Signature_should_be_as_expected` do the same, so I can remove that test !